src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -453,7 +446,71 @@ internal IEnumerable<Declaration> GetCompletionsForToken(ParseResult parseResult
             return Resolver.FindCompletions(
                 parseResult, parserLine, parserColumn, metadataDisplayInfoProvider);
         }
-        #endregion
 
+        /// <summary>
+        /// Wrapper method that calls Resolver.FindCompletions
+        /// </summary>
+        /// <param name="objectName"></param>
+        /// <param name="schemaName"></param>
+        /// <param name="objectType"></param>
+        /// <param name="tempFileName"></param>
+        /// <returns></returns>
+        internal ScriptingScriptOperation InitScriptOperation(string objectName, string schemaName, string objectType, string tempFileName)
+        {            
+            // object that has to be scripted
+            ScriptingObject obj = new ScriptingObject {
```

Comment:
>ScriptingObject [](start = 38, length = 15)

Any reason we don't reuse the objectmetadata class here?  We should generally use existing classes for a particular concept.  ObjectMetadata is for exactly the idea you're representing here.

Reply:
We have a special ScriptingObject class in sqltoolsservice, with method overrides for normal object base functions (such as equality etc). I assumed we should use the defined ScriptingObject class to be robust. Please let me know if you think otherwise.